src/util.ts

```diff
@@ -13,6 +14,289 @@ class Util {
             character: character
         } as Location;
     }
+
+    public identifyPlatform(platformCodeName: string): RokuHardwarePlatform {
+        const rokuPlatforms = this.getAllRokuPlatforms();
+        // Try an exact match
+        return rokuPlatforms.find(p => p.codeName.toLowerCase() === platformCodeName.toLowerCase()) ??
+            // Try a partial match
+            rokuPlatforms.find(p => p.codeName.toLowerCase().includes(platformCodeName.toLowerCase())) ??
+            // Split the code name and try an exact match with each part
+            rokuPlatforms.find(
+                p => platformCodeName.split(/;|,|_/) // Split on semicolon, comma, and underscore
+                    .filter(part => part.toLowerCase() !== '4k') // Ignore 4K indicator
+                    .some(part => p.codeName.toLowerCase().includes(part.toLowerCase()))
+            ) ??
+            // Return Unknown
+            {
+                productName: 'Unknown',
+                codeName: platformCodeName,
+                model: 'Unknown',
+                productionStatus: ProductionStatus.Unknown
+            };
+    }
+
+    // TODO: where to put this? should we store the codeName as string instead of RokuHardwarePlatform?
+    /**
+     * @see https://developer.roku.com/docs/specs/hardware.md
+    */
+    public getAllRokuPlatforms(): RokuHardwarePlatform[] {
```

Comment:
@chrisdp do you think it would be worth publishing a separate node_module with just this type of information? We could try to keep it up-to-date as new devices come out, and that way all of this type of information would be centralized in a single location.